src/adapters/collateralized_simple_interest_loan_adapter.ts

```diff
@@ -113,3 +134,214 @@ export class CollateralizedLoanTerms {
+            amortizationUnit,
+            termLength,
+        });
+
+        const packedCollateralizedParams = this.collateralizedLoanTerms.packParameters({
+            collateralTokenIndex,
+            collateralAmount,
+            gracePeriodInDays,
+        });
+
+        // Our final output is the perfect union of the packed simple interest params and the packed
+        // collateralized params.
+        const packedParams =
+            packedSimpleInterestParams.substr(0, 39) + packedCollateralizedParams.substr(39, 27);
+
+        debtOrder = {
+            ...debtOrder,
+            principalToken: principalToken.address,
+            termsContract: collateralizedSimpleInterestTermsContract.address,
+            termsContractParameters: packedParams,
+        };
+
+        return DebtOrder.applyNetworkDefaults(debtOrder, this.contractsAPI);
+    }
+
+    public async fromDebtOrder(
+        debtOrder: DebtOrder.Instance,
+    ): Promise<CollateralizedSimpleInterestLoanOrder> {
+        this.assert.schema.debtOrderWithTermsSpecified("debtOrder", debtOrder);
+
+        const {
+            principalTokenIndex,
+            principalAmount,
+            interestRate,
+            termLength,
+            amortizationUnit,
+        } = this.simpleInterestLoanTerms.unpackParameters(debtOrder.termsContractParameters);
+
+        const {
+            collateralTokenIndex,
+            collateralAmount,
+            gracePeriodInDays,
+        } = this.collateralizedLoanTerms.unpackParameters(debtOrder.termsContractParameters);
+
+        const principalTokenSymbol = await this.contractsAPI.getTokenSymbolByIndexAsync(
+            principalTokenIndex,
+        );
+
+        const collateralTokenSymbol = await this.contractsAPI.getTokenSymbolByIndexAsync(
+            collateralTokenIndex,
+        );
+
+        const collateralTokenAddress = await this.contractsAPI.getTokenAddressBySymbolAsync(
+            collateralTokenSymbol,
+        );
+
+        // Assert that the principal token corresponds to symbol we've unpacked.
+        this.assertTokenCorrespondsToSymbol(debtOrder.principalToken, principalTokenSymbol);
+
+        // Assert that the collateral token address corresponds to symbol we've unpacked.
+        this.assertTokenCorrespondsToSymbol(collateralTokenAddress, collateralTokenSymbol);
+
+        return {
+            ...debtOrder,
+            // simple interest terms.
+            principalAmount,
+            principalTokenSymbol,
+            interestRate,
+            termLength,
+            amortizationUnit,
+            // collateralized terms.
+            collateralTokenSymbol,
+            collateralAmount,
+            gracePeriodInDays,
+        };
+    }
+
+    public async fromDebtRegistryEntry(
+        entry: DebtRegistryEntry,
+    ): Promise<CollateralizedSimpleInterestLoanOrder> {
+        await this.assertIsCollateralizedSimpleInterestTermsContract(entry.termsContract);
+
+        const {
```

Comment:
All of this is shared between `fromDebtRegistryEntry` and `fromDebtOrder` -- can we create some sort of helper method to leverage for both -- i.e. `unpackParameters` on the `CollateralizedSimpleInterestLoanAdapter` class?

Reply:
Great call!